chromium_src/components/variations/study_filtering.cc

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "components/variations/study_filtering.h"
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+
+namespace variations::internal {
+
+namespace {
+
+class VersionStringWithMajorPartFromVersion {
+ public:
+  VersionStringWithMajorPartFromVersion(std::string_view version_string,
+                                        const base::Version& version)
+      : version_string_(version_string) {
+    if (!version.IsValid()) {
+      return;
+    }
+
+    auto splitted_version_string = base::SplitStringPiece(
+        version_string, ".", base::WhitespaceHandling::KEEP_WHITESPACE,
+        base::SplitResult::SPLIT_WANT_ALL);
+
+    // If the version filter is empty, "MAJOR" or "MAJOR.*", we compare it as
+    // is.
+    if (splitted_version_string.size() < 2 ||
+        (splitted_version_string.size() == 2 &&
+         splitted_version_string[1] == "*")) {
+      return;
+    }
+
+    // Otherwise set MAJOR in the filter to the MAJOR of the current version.
+    // This effectively skips the MAJOR part during version comparison, allowing
+    // to compare the version parts after the MAJOR (ex. 1.50.10) no matter what
+    // the MAJOR version is (ex. 130.1.50.10 or 131.1.50.10).
+    const std::string major_version_string =
+        base::NumberToString(version.components()[0]);
+    if (major_version_string == splitted_version_string[0]) {
+      return;
+    }
+
+    splitted_version_string[0] = major_version_string;
```

Comment:
1) Do we really want to apply this logic by default? I would rather use min/max versions like `0.1.72.4` or `1.72.4` for this case (and enforce this format in brave-variations).   
2) this PR changes how we apply the current seed, not only the studies. Is it okay?

Reply:
> 1. Do we really want to apply this logic by default? I would rather use min/max versions like `0.1.72.4` or `1.72.4` for this case (and enforce this format in brave-variations).

Supporting `0.1.72.4` is nice addition, but we can't do this by default as this will break existing installations. For ex., a new study with `min_version: '0.1.72.4'` filter will be enabled on all versions which don't support the `0.` special mode.

> 2. this PR changes how we apply the current seed, not only the studies. Is it okay?

Not sure I get this. Is there a seed version filter? The new logic should only be used when checking `Study` instances.